modules/util/portal-tools-rest-builder/src/main/java/com/liferay/portal/tools/rest/builder/internal/freemarker/tool/java/parser/ResourceOpenAPIParser.java

```diff
@@ -1366,6 +1369,27 @@ private static boolean _isValidParameter(String name, String schemaName) {
 		return true;
 	}
 
+	private static String _replaceDotAndUppercase(String inputString) {
+		int lastIndex = 0;
+		Matcher matcher = _dotPattern.matcher(inputString);
```

Comment:
I am not sure if using a regex here is the fastest/cleanest way to proceed

Reply:
```java
private static String _replaceDotAndUppercase(String inputString) {
	StringBuilder sb = new StringBuilder();

	for (int i = 0; i < inputString.length(); i++) {
		if (inputString.charAt(i) == '.') {
			sb.append("Dot");
			i++;

			if (i < inputString.length()) {
				sb.append(Character.toUpperCase(inputString.charAt(i)));
			}
		} 
		else {
			sb.append(inputString.charAt(i));
		}
	}

	return sb.toString();
}
```

Reply:
Do you prefer that?

Reply:
I think this could be even easier:

```
methodNameSegments.add(OpenAPIParserUtil.getHTTPMethod(operation));

path = StringUtil.replace(path, ".", "/Dot/"); // this is the new line

String[] pathSegments = path.split("/");
String pluralSchemaName = TextFormatter.formatPlural(schemaName);

```

Reply:
After the conversation we had online:

In my opinion `.Users` should behave the same as `MyUsers`. `.` is just a shortcut for the word `Dot`.

Reply:
We will leave it like this for now.